src/Nethermind/Nethermind.Blockchain/Blocks/BlockStore.cs

```diff
@@ -88,4 +89,9 @@ public void Cache(Block block)
     {
         _blockCache.Set(block.Hash, block);
     }
+
+    public Block[] GetAll()
+    {
+        return _blockDb.GetAllValues(true).Select(bytes => _blockDecoder.Decode(bytes.AsRlpStream())).ToArray();
+    }
```

Comment:
Potentially huge data set, a bit risky

Reply:
Change to IEnumerable and stream them instead of allocating an array

Reply:
I converted to IEnumerable so I think this is resolved but could you check? I'm not sure what you mean by streaming them - is this what an IEnumerable would do by default?

Reply:
yes IEnumerable is lazy evaluated, so it streams them